satpy/enhancements/__init__.py

```diff
@@ -120,6 +128,65 @@ def func(band_data, index=None):
     return apply_enhancement(img.data, func, separate=True)
 
 
+def interp_scaling(
+        img: XRImage,
+        xp: ArrayLike,
+        fp: ArrayLike,
+        coordinate_divisor: Number = None,
+        **kwargs):
+    """Apply 1D linear interpolation.
+
+    This uses :func:`numpy.interp` mapped over the provided dask array chunks.
+
+    Args:
+        img: Image data to be scaled. It is assumed the data is already
+            normalized between 0 and 1.
+        xp: X coordinates of the image data points used for interpolation.
+            This is passed directly to :func:`numpy.interp`.
+        fp: Y coordinate of the output image data poitns used for
+            interpolation. This is passed directly to :func:`numpy.interp`.
+        coordinate_divisor: Divide ``xp`` and ``fp`` by this value before
+            passing using for interpolation. This is a convenience to make
+            matching normalized image data to interp coordinates or to avoid
+            floating point precision errors in YAML configuration files.
+            If not provided, ``xp`` and ``fp`` will not be modified.
+
+    Examples:
+        YAML example:
+
+        .. code-block:: yaml
+
+              true_color_linear_interpolation:
+                sensor: abi
+                standard_name: true_color
+                operations:
+                - name: reflectance_range
+                  method: !!python/name:satpy.enhancements.stretch
+                  kwargs: {stretch: 'crude', min_stretch: 0., max_stretch: 100.}
+                - name: Linear interpolation
+                  method: !!python/name:satpy.enhancements.interp_scaling
+                  kwargs:
+                   xp: [0., 25., 55., 100., 255.]
+                   fp: [0., 90., 140., 175., 255.]
+                   coordinate_divisor: 255
+
+    """
+    LOG.debug("Applying the interp_scaling")
+    if coordinate_divisor is not None:
+        xp = np.asarray(xp) / coordinate_divisor
+        fp = np.asarray(fp) / coordinate_divisor
+
+    def func(band_data, xp, fp, index=None):
+        # Interpolate band on [0,1] using "lazy" arrays (put calculations off until the end).
+        band_data = xr.DataArray(da.clip(band_data.data.map_blocks(np.interp, xp=xp, fp=fp), 0, 1),
+                                 coords=band_data.coords, dims=band_data.dims, name=band_data.name,
+                                 attrs=band_data.attrs)
+        return band_data
+
+    return apply_enhancement(img.data, func, separate=True,
+                             func_kwargs={'xp': xp, 'fp': fp})
```

Comment:
yeah, looks like a partial would work...

Reply:
I'd like to avoid partials...although if dask can serialize them maybe it doesn't matter. Dask is not currently doing anything with these functions but I could see it being done in the future. Do you think a partial makes this cleaner?

Reply:
Oh, can you make a partial with the args as I have them? The caller of the partial needs to provide the first argument.

Reply:
Yes, kwargs can be provided to the partial

Reply:
Using a partial would avoid passing `func_kwargs` down the `apply_enhancement` method...